daemon/logger/jsonfile/utils.go

```diff
@@ -193,15 +193,19 @@ func seekOffsetByTailLines(rs io.ReadSeeker, n int) (int64, error) {
 		cnt   = 0
 		left  = int64(0)
 		b     []byte
+
+		readN = int64(0)
 	)
 
 	for {
+		readN = int64(blockSize)
 		left = size + int64(block*blockSize)
 		if left < 0 {
+			readN = int64(blockSize) + left
 			left = 0
 		}
 
-		b = make([]byte, blockSize)
+		b = make([]byte, readN)
```

Comment:
If the content don't end with `\n` like `1\n2\n3\n4`, `seekOffsetByTailLines(, 2)` will return the index of `2`. The tail content would be `2\n3\n4`. Compared with command tail, `echo "1\n2\n3\n4" > a.txt; tail -n2 a.txt` is `3\n4`.
Does this function works as expected?😁 @fuweid